inertia/tests/test_deferred_prop.py

```diff
@@ -0,0 +1,131 @@
+import asyncio
+from fastapi import FastAPI, Depends
+from typing import Annotated
+
+from starlette.testclient import TestClient
+
+from inertia import (
+    Inertia,
+    inertia_dependency_factory,
+    InertiaResponse,
+    InertiaConfig,
+    defer,
+)
+from .utils import templates
+
+
+app = FastAPI()
+
+InertiaDep = Annotated[
+    Inertia, Depends(inertia_dependency_factory(InertiaConfig(templates=templates)))
+]
+
+
+async def get_users_data() -> dict[str, str]:
+    # Simulate a data fetch that takes some time to retrieve
+    await asyncio.sleep(0.1)
```

Comment:
nit: no need to actually wait in tests, it just adds to execution time.
You can probably mock a clock if you want